libvast/vast/system/schema_store.hpp

```diff
@@ -0,0 +1,49 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/schema.hpp"
+#include "vast/system/accountant.hpp"
+#include "vast/system/atoms.hpp"
+
+#include <caf/expected.hpp>
+#include <caf/typed_actor.hpp>
+#include <caf/typed_event_based_actor.hpp>
+
+namespace vast::system {
+
+// clang-format off
+using schema_store_type = caf::typed_actor<
+  caf::reacts_to<telemetry_atom>,
+  caf::replies_to<status_atom>::with<caf::dictionary<caf::config_value>>,
+  caf::reacts_to<caf::stream<table_slice_ptr>>,
+  caf::replies_to<get_atom>::with<schema>
+>;
+// clang-format on
+
+struct schema_store_state {
+  void send_report() const;
+
+  schema_store_type::stateful_pointer<schema_store_state> self;
+  accountant_type accountant;
+
+  schema data;
```

Comment:
For the time being, we don't need a `schema get()` handler that returns everything. You can modify that to take a string for the lookup and wrap `equal_range` and return a `vector<type>`.

Reply:
The get handler must still return the name as well, right?

So it's gonna be a multimap of name to type, and a get function that returns a schema (because that is a named vector of types, exactly).